kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/client/grpc/GrpcRS3Client.java

```diff
@@ -391,10 +393,25 @@ public CreateStoreResult createStore(
     return new CreateStoreResult(uuidFromProto(result.getStoreId()), result.getPssIdsList());
   }
 
-  private void checkField(final Supplier<Boolean> check, final String field) {
-    if (!check.get()) {
-      throw new RuntimeException("rs3 resp proto missing field " + field);
+  @Override
+  public PssCheckpoint createCheckpoint(
+      final UUID storeId,
+      final LssId lssId,
+      final int pssId,
+      final Optional<Long> expectedWrittenOffset) {
+    final var request = Rs3.CreateCheckpointRequest.newBuilder()
+        .setLssId(lssIdProto(lssId))
+        .setExpectedWrittenOffset(walOffsetProto(expectedWrittenOffset))
+        .build();
+    final RS3Grpc.RS3BlockingStub stub = stubs.stubs(storeId, pssId).syncStub();
+    final Rs3.CreateCheckpointResult result;
+    try {
```

Comment:
We could use `withRetry`?

Reply:
Yeah, it should be idempotent so that makes sense